### ISSUE TITLE
Source path problem in json and html reporter

### ISSUE THREAD

**Issue by karin-exini** [mattlewis92/karma-coverage-istanbul-reporter#9]
Pathes to source ts files in json, json-summary and html reporters is wrong. Path ends up being `C:/<path to project>/C:\<path to project>\client\modules\app\app-routing.module.ts`. A file with that name is of course not found.

I am using windows, is this a windows problem? 

My setup is shown below. What am I doing wrong? 

I also want to note that paths are correct with `istanbul-instrumenter-loader` and `remap-istanbul`, so I suspect that I am missing some setup in the `karma-coverage-istanbul-reporter`.

karma.config.js
```
const path = require('path');
const webpack = require('webpack');

module.exports = function (config) {
    var _config = {
        basePath: '',

        frameworks: ['jasmine'],

        files: [
            { pattern: './node_modules/jquery/dist/jquery.js', watched: false},
            { pattern: './node_modules/materialize-css/dist/js/materialize.js', watched: false},
            { pattern: './karma-test-shim.js', watched: false}
        ],

        preprocessors: {
            './karma-test-shim.js': ['webpack', 'sourcemap']
        },

        webpack: require('./webpack.test.config.js'),

        reporters: ['progress', 'coverage-istanbul'],
        port: 9876,
        colors: true,
        logLevel: config.LOG_DEBUG,
        autoWatch: false,
        browsers: ['PhantomJS'],
        client: {
            captureConsole: true
        },
        singleRun: true,

        // Webpack std out off
        webpackServer: { noInfo: true },

        phantomjsLauncher: {
            // Have phantomjs exit if a ResourceError is encountered (useful if karma exits without killing phantom)
            exitOnResourceError: true
        },

        coverageIstanbulReporter: {
            reports: [ 'html', 'text-summary', 'json', 'json-summary', 'text' ],
            fixWebpackSourcePaths: true,
            dir: path.join(__dirname, 'coverage')
        }
    };

    config.set(_config);
};

```

karma-test-shim.js
```
Error.stackTraceLimit = Infinity;

// Load scripts needed by angular
require('core-js/es6');
require('core-js/es7/reflect');
require('reflect-metadata');
require('zone.js/dist/zone');
require('zone.js/dist/long-stack-trace-zone');
require('zone.js/dist/proxy');
require('zone.js/dist/sync-test');
require('zone.js/dist/jasmine-patch');
require('zone.js/dist/async-test');
require('zone.js/dist/fake-async-test');

// Load RxJS
require('rxjs/Rx');

// Load all ts files for tests
var testContext = require.context('./client', true, /\.spec\.ts$/);
testContext.keys().forEach(testContext);

// Load all components (for coverage)
var componentContext = require.context('./client/modules', true, /\.ts$/);
componentContext.keys().forEach(componentContext);

// Wire upp angular testing framework
var testing = require('@angular/core/testing');
var browser = require('@angular/platform-browser-dynamic/testing');

testing.TestBed.initTestEnvironment(browser.BrowserDynamicTestingModule, browser.platformBrowserDynamicTesting());

```

webpack.test.config.js
```
var path = require('path'),
    webpack = require('webpack');

module.exports = {
    devtool: 'inline-source-map',

    resolve: {
        root: path.resolve(__dirname, 'client'),
        modulesDirectories: ['node_modules'],
        extensions: ['', '.ts', '.js', '.json', '.css', '.html'],
        alias: {
            materializecss: path.resolve(__dirname, 'node_modules/materialize-css/dist/css/materialize.css'),
            materialize: path.resolve(__dirname, 'node_modules/materialize-css/dist/js/materialize.js')
        }
    },

    module: {
        preLoaders: [
            {
                test: /\.min\.js$/,
                loader: 'source-map'
            },
            {
                test: /\.ts$/,
                loader: 'tslint',
                exclude: [/node_modules/]
            }
        ],
        loaders: [
            {
                test: /\.ts$/,
                loaders: [
                    'awesome-typescript-loader?sourceMap',
                    'angular2-template-loader',
                    'angular-router-loader'
                ],
                exclude: [/node_modules/]
            },
            {
                test: /\.css$/,
                loader: 'skip'
            },
            {
                test: /materialize-css\/dist\/js\/materialize\.js/,
                loader: 'imports?materializecss'
            },
            {
                test: /\.woff2?$|\.ttf$|\.eot$|\.svg$/,
                loader: 'null'
            },
            {
                test: /.(png|woff(2)?|eot|ttf|svg)(\?[a-z0-9=\.]+)?$/,
                loader: 'null',
                exclude: /node_modules/
            },
            {
                test: /\.html$/,
                loader: 'html'
            },
            {
                test: /\.json$/,
                loader: 'json'
            }
        ],
        postLoaders: [
            {
                test: /\.ts$/,
                loader: 'istanbul-instrumenter-loader',
                exclude: [
                    'node_modules',
                    /\.spec\.ts$/
                ]
            }
        ]
    },

    tslint: {
        emitErrors: true,
        failOnHint: false
    }
};
```

**Comment by mattlewis92** [comment]
I suspect this is due to a configuration error somewhere, but I can't see anything wrong with the config you provided, would it be possible for you to share a minimal runnable repo that reproduces your issue? Thanks!

**Comment by karin-exini** [comment]
Thank you for your response. A minimal repo for reproducing the behavior is located at https://github.com/karin-exini/coverage-istanbul-reporter-path
Running `npm test` produces html reports in the `coverage` directory with the path error. The reports I got on my machine are included in the repo. 
As a note, I use Windows 10 and npm version 3.10.10.

**Comment by mattlewis92** [comment]
Thanks for that! It does appear to be a windows issue as it works for me on my mac. I will dig into it and push out a fix asap.

**Comment by mattlewis92** [comment]
I tried to dig into this today but sadly couldn't make any progress due to windows permission errors whenever I tried to run the tests. @karin-exini as you develop on windows full time maybe you might have better luck than me at fixing this? If you clone this repo, `npm install` and then `npm test` and then manage to fix the tests on windows and could send a PR with a fix that would be a great help! 😄

**Comment by karin-exini** [comment]
Thank you for your time, I will look into this and get back to you.

**Comment by karin-exini** [comment]
I looked into this, and I can't run karma with `karma.Server` on windows either. But I managed to do some digging in my test project. The problem seems to be mix-up between windows and unix style pathes. But if the problems lies in your plugin or someone else's, I cannot say. 

Anyway, in your plugin the object `fileCoverage.inputSourceMap` look like this:
```
 { version: 3,
  file: 'C:/development/git/coverage-istanbul-reporter-path/client/modules/app/app.component.ts',
  sourceRoot: 'C:/development/git/coverage-istanbul-reporter-path/',
  sources: [ 'C:\\development\\git\\coverage-istanbul-reporter-path\\client\\modules\\app\\app.component.ts' ],
  names: [],
  mappings: ';;;;;;;AAAA,sCAAyC;AAQzC,IAAa,YAAY;IAAzB;IACA,CAAC;IAAD,mBAAC;AAAD,CAAC,AADD,IACC;AADY,YAAY;IANxB,gBAAS,CAAC;QACP,QAAQ,EAAE,UAAU;QACpB,QAAQ,EAAE,iCAET;KACJ,CAAC;GACW,YAAY,CACxB;AADY,oCAAY',
  sourcesContent: [ 'import { Component } from "@angular/core"\r\n\r\n@Component({\r\n    selector: "test-app",\r\n    template: `\r\n        <p>Test app</p>\r\n    `\r\n})\r\nexport class AppComponent {\r\n}\r\n' ] }
```
After `sourceMapStore.transformCoverage(coverageMap).map` <a href="https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/master/src/reporter.js#L58">here</a> the `remappedCoverageMap` looks like this: 
```
 CoverageMap {
  data: { 'C:/development/git/coverage-istanbul-reporter-path/C:\development\git\coverage-istanbul-reporter-path\client\modules\app\app.component.ts': MappedCoverage { data: [Object], meta: [Object] } } }
```

The problem is that `source-map` plugin can not handle that `sources` and `sourceRoot` in `inputSourceMap` is of different path-style. Way down in that plugin the problem arises <a href="https://github.com/mozilla/source-map/blob/master/lib/source-map-consumer.js#L621">here</a>.

Do you have any thoughts on this?

**Comment by mattlewis92** [comment]
Ah that would explain it, it seems like webpack is messing up the sourcemap paths. Would you mind trying the PR I just opened to see if it fixes it for you?

```
npm i -D mattlewis92/karma-coverage-istanbul-reporter#windows-sources-fix
```

**Comment by mattlewis92** [comment]
Should be fixed now in 0.3.1 🎉

**Comment by karin-exini** [comment]
It works, perfect, thanks!
